libs/agno/agno/os/app.py

```diff
@@ -92,80 +92,55 @@ def __init__(
-                if entity.db.session_table_name:
-                    if _add_unique_component("session", f"{entity.db.session_table_name}"):
-                        discovered_apps.append(SessionApp(db=entity.db, display_name=entity.db.session_table_name))
-
-                # Metrics app
-                if entity.db.metrics_table_name:
-                    if _add_unique_component("metrics", f"{entity.db.metrics_table_name}"):
-                        discovered_apps.append(MetricsApp(db=entity.db, display_name=entity.db.metrics_table_name))
-
-                # Eval app
-                if entity.db.eval_table_name:
-                    if _add_unique_component("eval", f"{entity.db.eval_table_name}"):
-                        discovered_apps.append(EvalApp(db=entity.db, display_name=entity.db.eval_table_name))
-
-            # Knowledge app
-            if entity.knowledge:
-                if not entity.knowledge.contents_db:
-                    log_warning("Knowledge contents_db is required to use knowledge inside AgentOS.")
-                    return []
-
-                db = entity.knowledge.contents_db
-                if _add_unique_component("knowledge", f"{db.knowledge_table_name}") and entity.knowledge:
-                    discovered_apps.append(
-                        KnowledgeApp(knowledge=entity.knowledge, display_name=db.knowledge_table_name)
-                    )
-
-            return discovered_apps
-
-        # Process agents
-        if self.agents:
-            for agent in self.agents:
-                _auto_discover_entity_apps(agent)
+    def _load_yaml_config(self, config_file_path: str) -> AgentOSConfig:
+        """Load a YAML config file and return the configuration as an AgentOSConfig instance."""
+        from pathlib import Path
 
-        # Process teams
-        if self.teams:
-            for team in self.teams:
-                _auto_discover_entity_apps(team)
-                for member in team.members:
-                    _auto_discover_entity_apps(member)
+        import yaml
+
+        # Validate that the path points to a YAML file
+        path = Path(config_file_path)
+        if path.suffix.lower() not in [".yaml", ".yml"]:
+            raise ValueError(f"Config file must have a .yaml or .yml extension, got: {config_file_path}")
+
+        # Load the YAML file
+        with open(config_file_path, "r") as f:
+            return AgentOSConfig.model_validate(yaml.safe_load(f))
 
-        # Process workflows
-        # TODO: Implement workflow app discovery
+    def _auto_discover_databases(self) -> None:
+        """Auto-discover the databases used by all contextual agents, teams and workflows."""
+        dbs = {}
 
-        # Log discovered apps
-        if discovered_apps:
-            log_debug(f"Apps added to AgentOS: {[app.display_name for app in discovered_apps]}")
+        for agent in self.agents or []:
+            if agent.db:
+                dbs[agent.db.id] = agent.db
```

Comment:
I don't think these IDs are auto-generated right? They should be.

Reply:
Fair enough, let's do that